nautobot_netbox_importer/management/commands/import_netbox_objectchange_json.py

```diff
@@ -0,0 +1,144 @@
+        """Add parser arguments to the import_netbox_objectchange_json management command."""
+        parser.add_argument(
+            "json_file",
+            type=argparse.FileType("r"),
+            help=(
+                "DB dump in JSON without ObjectChange objects. "
+                "Indeed, the same used in the 'import_netbox_json' command."
+            ),
+        )
+        parser.add_argument(
+            "objectchange_json_file",
+            type=argparse.FileType("r"),
+            help=("DB dump in JSON with ONLY ObjectChange objects. "),
+        )
+        parser.add_argument("netbox_version", type=version.parse)
+        parser.add_argument("--dry-run", action="store_true", default=False)
+
+    def map_object_type(self, key, entry, used_error_messages):
+        """Map a ContentType object from Netbox to Nautobot. Returns False if not possible."""
+        app_label, modelname = self.netbox_contenttype_mapping[entry["fields"][key]]
+        try:
+            contenttype_id = self.nautobot_contenttype_mapping[(app_label, modelname)]
+            entry["fields"][key] = ContentType.objects.get(id=contenttype_id)
+        except KeyError:
+            error_message = f"{(app_label, modelname)} key is not present in Nautobot"
+            if error_message not in used_error_messages:
+                used_error_messages.add(error_message)
+                self.logger.warning(error_message)
+            return False
+        return True
+
+    def process_objectchange(self, entry, options, used_error_messages):
+        """Processes one ObjectChange entry (dict) to migrate from Netbox to Nautobot."""
+        if not self.map_object_type("changed_object_type", entry, used_error_messages):
+            return
+
+        if entry["fields"]["related_object_type"]:
+            if not self.map_object_type("related_object_type", entry, used_error_messages):
+                return
+        else:
+            del entry["fields"]["related_object_type"]
+
+        modelname = entry["fields"]["changed_object_type"].model
+        entry["fields"]["changed_object_id"] = netbox_pk_to_nautobot_pk(modelname, entry["fields"]["changed_object_id"])
+        if entry["fields"]["related_object_id"]:
+            modelname = entry["fields"]["related_object_type"].model
+            entry["fields"]["related_object_id"] = netbox_pk_to_nautobot_pk(
+                modelname, entry["fields"]["related_object_id"]
+            )
+
+        entry["fields"]["user"] = User.objects.filter(username=entry["fields"]["user_name"]).first()
+        if not entry["fields"]["user"]:
+            error_message = f'Username {entry["fields"]["user_name"]} not present in DB.'
+            if error_message not in used_error_messages:
+                used_error_messages.add(error_message)
+                self.logger.error(error_message)
+            return
+
+        if not options["dry_run"]:
+            obj = ObjectChange.objects.create(**entry["fields"])
```

Comment:
It occurs to me, should this command be idempotent (running it more than once shouldn't result in the creation of duplicate/redundant ObjectChange records)?

Reply:
yup, it should